lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_scaleset.py

```diff
@@ -595,6 +617,10 @@ def exec_module(self, **kwargs):
 
                     managed_disk = self.compute_models.VirtualMachineScaleSetManagedDiskParameters(storage_account_type=self.managed_disk_type)
 
+                    network_security_group = None
+                    if self.network_security_group:
+                        network_security_group = self.get_network_security_group(self.network_security_group)
```

Comment:
since `self.network_security_group_resource_group` is not required, I suggest add a logic to make `self.network_security_group_resource_group = self.network_security_group_resource_group or self.resource_group`, then change the `get_network_security_group(self, name)` to `get_network_security_group(self, resource_group, name)`

Reply:
Hi @yuwzho, as per the suggestion I have modified the code to include resource group as a parameter to get_network_security_group method. Please review and let me know if you would like to have any more improvements.